sources/updater/updog/src/main.rs

```diff
@@ -453,6 +456,49 @@ fn initiate_reboot() -> Result<()> {
     Ok(())
 }
 
+/// Our underlying HTTP client, reqwest, supports proxies by reading the HTTPS_PROXY and NO_PROXY
+/// environment variables. Bottlerocket services can source proxy.env before running, but updog is
+/// not a service, so we read these values from the config file and add them to the environment
+/// here.
+fn set_https_proxy_environment_variables(
+    https_proxy: &Option<String>,
+    no_proxy: &Option<Vec<String>>,
+) -> Result<()> {
+    let proxy = match https_proxy {
+        Some(s) if !s.is_empty() => s.to_owned(),
+        // without https_proxy, no_proxy does nothing, so we are done
+        _ => return Ok(()),
+    };
+
+    // TODO - remove this if reqwest adds support for proxy addresses with no scheme
```

Comment:
I think you opened a GitHub issue for this in reqwest's repository, want to link it here?

Reply:
Linked to our own #1332 since it's a little bit wrapped up in a dependency knot.